ocaml/xapi/test_vdi_allowed_operations.ml

```diff
@@ -262,6 +270,25 @@ let test_cbt =
       )
   in
 
+  let test_operations_are_allowed_on_snapshot_vdis_attached_to_vm_snapshots = for_vdi_operations
+      [`data_destroy; `list_changed_blocks]
+      (fun op ->
+         let __context = Mock.make_context_with_new_db "Mock context" in
+         let vM = Test_common.make_vm ~__context ~is_a_template:true () in
+         (* Set up the fields corresponding to a VM snapshot *)
+         Db.VM.set_is_a_snapshot ~__context ~self:vM ~value:true;
+         Db.VM.set_power_state ~__context ~self:vM ~value:`Suspended;
+         run_assert_equal_with_vdi
+           ~__context
+           ~vdi_fun:(fun vDI ->
+               let (_: API.ref_VBD) = Test_common.make_vbd ~__context ~vM ~vDI ~currently_attached:true () in
+               Db.VDI.set_cbt_enabled ~__context ~self:vDI ~value:true;
+               Db.VDI.set_is_a_snapshot ~__context ~self:vDI ~value:true)
+           op
+           None
+      )
+  in
+
```

Comment:
Is there some other test that tests that these operations are forbidden if the VM is not a snapshot?

Reply:
No, due to lack of time, but they are planned. https://github.com/xapi-project/xen-api/pull/3211 adds a test for data_destroy, a unit test for list_changed_blocks is planned.